ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/dns/DNSResolver.java

```diff
@@ -151,21 +147,22 @@ private boolean internalVisit(
     }
 
     final DNSEntry entry = optionalDNSEntry.get();
-    if (entry instanceof ENRNode node) {
-      // TODO: this always return true because the visitor is reference to list.add
-      return visitor.visit(node.nodeRecord());
-    } else if (entry instanceof DNSEntry.ENRTree tree) {
-      for (String e : tree.entries()) {
-        // TODO: When would this ever return false?
-        boolean keepGoing = internalVisit(e, domainName, visitor);
-        if (!keepGoing) {
-          return false;
+    switch (entry) {
+      case ENRNode node -> {
+        // TODO: this always return true because the visitor is reference to list.add
+        return visitor.visit(node.nodeRecord());
+      }
+      case DNSEntry.ENRTree tree -> {
+        for (String e : tree.entries()) {
+          // TODO: When would this ever return false?
```

Comment:
could you do something about these TODO?

Reply:
These were leftover questions for myself after I "adapted" code from Tuweni, I'll remove these comments :).